src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/jvmsizing/HighOldGenOccupancyRca.java

```diff
@@ -0,0 +1,147 @@
+  private static final long EVAL_INTERVAL_IN_S = 5;
+  private static final int B_TO_MB = 1024 * 1024;
+
+  private final Metric heapUsed;
+  private final Metric heapMax;
+  private final SlidingWindow<SlidingWindowData> oldGenUtilizationSlidingWindow;
+
+  private long heapUtilizationThreshold;
+  private long rcaEvaluationIntervalInS;
+  private long rcaSamplesBeforeEval;
+  private long samples;
+
+  /**
+   * Create HighOldGenOccupancyRca with default values.
+   * @param heapMax The heapMax metric.
+   * @param heapUsed The heapUsed metric.
+   */
+  public HighOldGenOccupancyRca(final Metric heapMax, final Metric heapUsed) {
+    this(heapMax, heapUsed, HighOldGenOccupancyRcaConfig.DEFAULT_UTILIZATION,
+        HighOldGenOccupancyRcaConfig.DEFAULT_EVALUATION_INTERVAL_IN_S);
+  }
+
+  public HighOldGenOccupancyRca(final Metric heapMax, final Metric heapUsed,
+      final long heapUtilizationThreshold, final long rcaEvaluationIntervalInS) {
+    super(EVAL_INTERVAL_IN_S);
+    this.oldGenUtilizationSlidingWindow = new SlidingWindow<>(1, TimeUnit.MINUTES);
+    this.heapUsed = heapUsed;
+    this.heapMax = heapMax;
+    this.heapUtilizationThreshold = heapUtilizationThreshold;
+    this.rcaEvaluationIntervalInS = rcaEvaluationIntervalInS;
+    this.rcaSamplesBeforeEval = rcaEvaluationIntervalInS / EVAL_INTERVAL_IN_S;
+    this.samples = 0;
+  }
+
+  @Override
+  public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+    throw new UnsupportedOperationException("generateFlowUnitListFromWire should not be called "
+        + "for node-local rca: " + args.getNode().name());
+  }
+
+  @Override
+  public ResourceFlowUnit<HotResourceSummary> operate() {
+    samples++;
+    addToSlidingWindow();
+    if (samples == rcaSamplesBeforeEval) {
+      samples = 0;
+      return evaluateAndEmit();
+    }
+
+    return new ResourceFlowUnit<>(System.currentTimeMillis());
+  }
+
+  private ResourceFlowUnit<HotResourceSummary> evaluateAndEmit() {
+    long currTime = System.currentTimeMillis();
+    double averageUtilizationPercentage = oldGenUtilizationSlidingWindow.readAvg();
+    ResourceContext context = new ResourceContext(State.HEALTHY);
+    HotResourceSummary summary = new HotResourceSummary(ResourceUtil.OLD_GEN_HEAP_USAGE,
+        (double)heapUtilizationThreshold,
+        averageUtilizationPercentage, (int)rcaEvaluationIntervalInS);
+    if (averageUtilizationPercentage >= heapUtilizationThreshold) {
+      context = new ResourceContext(State.UNHEALTHY);
+    }
+    return new ResourceFlowUnit<>(currTime, context, summary);
+  }
+
+  private void addToSlidingWindow() {
+    double oldGenUsed = getOldGenValueForMetric(heapUsed);
+    double maxOldGen = getOldGenValueForMetric(heapMax);
+
+    if (maxOldGen == 0d) {
+      LOG.info("Max Old Gen capacity cannot be 0. Skipping.");
+    }
+
+    this.oldGenUtilizationSlidingWindow.next(new SlidingWindowData(System.currentTimeMillis(),
+        oldGenUsed / maxOldGen * 100d));
```

Comment:
nit: parentheses for order of operations instead of relying on L to R evaluation

Reply:
+1. do we really need the *100 here or should we just work with decimals ( [0,1] values)?

Reply:
done, but kept the 100 just so that percentage looks normal. This was just a personal preference.